service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceMethod.java

```diff
@@ -178,6 +181,7 @@ private synchronized MethodResult invokeMethod(URL url, String methodName, Long
         MethodParser methodParserResponse = new MethodParser();
         methodParserResponse.fromJson(new String(response.getBody(), StandardCharsets.UTF_8));
 
+        log.debug("After invoke: url={}, methodName={}, connectTimeout={} seconds, responseTimeout={} seconds", url, methodName, connectTimeoutInSeconds, responseTimeoutInSeconds);
```

Comment:
>log [](start = 8, length = 3)

Is there a good way to correlate this back to the log above? What if service client invokes the same method on the same device twice from two different threads

Reply:
Good point. Added `requestId` in the log.